src/components/Header/Header.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  FC, FormEvent,
+  memo, useContext, useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { AuthContext } from '../Auth/AuthContext';
+
+type Props = {
+  isAddingTodo: boolean,
+  showError: (message: string) => void,
+  onAddTodo: (fieldsForCreate: Omit<Todo, 'id'>) => Promise<void>;
+  handleToggleTodosStatus: () => void;
+  shouldRenderActiveToggle: boolean,
+};
+
+export const Header: FC<Props> = memo((props) => {
+  const {
+    onAddTodo,
+    showError,
+    shouldRenderActiveToggle,
+    isAddingTodo,
+    handleToggleTodosStatus,
+  } = props;
+
+  const user = useContext(AuthContext);
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const [title, setTitle] = useState('');
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [isAddingTodo]);
+
+  const onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      showError('Title is required');
+
+      return;
+    }
+
+    if (!user) {
+      showError('User not found');
+
+      return;
+    }
+
+    try {
+      await onAddTodo({
+        title,
+        userId: user.id,
```

Comment:
You should create `onAddTodo` function that doesn't take userId, but take it from the closure.
So you don't need to pass it here